From 3352901fd5ea4bb31f3727f48e84dca668717258 Mon Sep 17 00:00:00 2001
From: Scott Moser <scott.moser@chainguard.dev>
Date: Tue, 22 Oct 2024 10:20:28 -0400
Subject: [PATCH] If productname is LibreOfficeDev use double-dash args.

This reduces noise in ouput.  Our libreoffice gets built
with a 'productname' of LibreOfficeDev.  Maybe that should
be fixed, but this is fine here.

Also allows us to drop distutils import as we are never going
to run with a version of libreoffice < 3.3.

Running with '-vvv' will show the productname:

    $ unoconv -vvv my.csv
    Verbosity set to level 3
    Using office base path: /usr/lib64/libreoffice
    Using office binary path: /usr/lib/libreoffice/program
    DEBUG: Connection type: socket,host=127.0.0.1,....
    DEBUG: Existing listener not found.
    DEBUG: Launching our own listener using .../program/soffice.bin.
    LibreOfficeDev listener arguments are [...

We're changing to use '--headless' rather than '-headless' (and other
single-dash flavor args).
---
 unoconv | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/unoconv b/unoconv
index cce79f4..34843d7 100644
--- a/unoconv
+++ b/unoconv
@@ -16,7 +16,6 @@
 
 from __future__ import print_function
 
-from distutils.version import LooseVersion
 import getopt
 import glob
 import os
@@ -867,10 +866,7 @@ class Convertor:
             info(3, "Launching our own listener using %s." % office.binary)
             try:
                 product = self.svcmgr.createInstance("com.sun.star.configuration.ConfigurationProvider").createInstanceWithArguments("com.sun.star.configuration.ConfigurationAccess", UnoProps(nodepath="/org.openoffice.Setup/Product"))
-                if product.ooName not in ('LibreOffice', 'LOdev') or LooseVersion(product.ooSetupVersion) <= LooseVersion('3.3'):
-                    args = [office.binary, "-headless", "-invisible", "-nocrashreport", "-nodefault", "-nofirststartwizard", "-nologo", "-norestore", "-accept=%s" % op.connection]
-                else:
-                    args = [office.binary, "--headless", "--invisible", "--nocrashreport", "--nodefault", "--nofirststartwizard", "--nologo", "--norestore", "--accept=%s" % op.connection]
+                args = [office.binary, "--headless", "--invisible", "--nocrashreport", "--nodefault", "--nofirststartwizard", "--nologo", "--norestore", "--accept=%s" % op.connection]
                 if op.userProfile:
                     args.append("-env:UserInstallation=file://" + realpath(op.userProfile))
                 info(2, '%s listener arguments are %s.' % (product.ooName, args))
@@ -1235,10 +1231,7 @@ class Listener:
             else:
                 info(1, "Existing %s listener found, nothing to do." % product.ooName)
                 return
-            if product.ooName != "LibreOffice" or LooseVersion(product.ooSetupVersion) <= LooseVersion('3.3'):
-                cmd = [office.binary, "-headless", "-invisible", "-nocrashreport", "-nodefault", "-nologo", "-nofirststartwizard", "-norestore", "-accept=%s" % op.connection]
-            else:
-                cmd = [office.binary, "--headless", "--invisible", "--nocrashreport", "--nodefault", "--nologo", "--nofirststartwizard", "--norestore", "--accept=%s" % op.connection]
+            cmd = [office.binary, "--headless", "--invisible", "--nocrashreport", "--nodefault", "--nologo", "--nofirststartwizard", "--norestore", "--accept=%s" % op.connection]
 
             # The rationale for using subprocess.Popen is to be able to handle
             # a SIGTERM signal below and properly terminate the started office
@@ -1449,4 +1442,4 @@ if __name__ == '__main__':
         main()
     except KeyboardInterrupt:
         die(6, 'Exiting on user request')
-    die(exitcode)
\ No newline at end of file
+    die(exitcode)
-- 
2.47.0

